### PR TITLE
Move identical network functions to receive.pm

### PR DESCRIPTION
**Pull request by Henrybk**
Mostly identical functions, sub changeToInGameState was also removed from both servertype0 and sakexe_0.

vender_found had a difference in the hook calls.

That is all I remember that was different.

### CODE REVIEW

File: src/Network/Receive/ServerType0.pm [hunk 1]
```diff
@@ -1976,48 +1732,6 @@ sub npc_talk {
...
-	$pet{renameflag} = $args->{renameflag};
-	$pet{level} = $args->{level};
-	$pet{hungry} = $args->{hungry};
-	$pet{friendly} = $args->{friendly};
-	$pet{accessory} = $args->{accessory};
-	$pet{type} = $args->{type} if (exists $args->{type});
-	debug "Pet status: name=$pet{name} name_set=". ($pet{renameflag} ? 'yes' : 'no') ." level=$pet{level} hungry=$pet{hungry} intimacy=$pet{friendly} accessory=".itemNameSimple($pet{accessory})." type=".($pet{type}||"N/A")."\n", "pet";
-}
-
 sub pet_info2 {
```
Comment (lututui) [inline src/Network/Receive/ServerType0.pm]: pet_info2 was removed from Sakexe_0 and added to Receive, but it's still in ServerType0

File: src/Network/Receive/ServerType0.pm [hunk 2]
```diff
@@ -2861,143 +2270,6 @@ sub character_equip {
 	message($msg, "list");
 }
 
-sub storage_password_request {
```
Comment (lututui) [inline src/Network/Receive/ServerType0.pm]: storage_password_request was removed from ServerType0 and added to Receive, but it's still in Sakexe_0

File: src/Network/Receive/ServerType0.pm [hunk 3]
```diff
@@ -2861,143 +2270,6 @@ sub character_equip {
...
-		}
-	} else {
-		debug(($args->{switch} eq '023E') ?
-			"Character password: unknown flag $args->{flag}\n" :
-			"Storage password: unknown flag $args->{flag}\n");
-	}
-}
-
-# TODO
-sub storage_password_result {
```
Comment (lututui) [inline src/Network/Receive/ServerType0.pm]: storage_password_result was removed from ServerType0 and added to Receive, but it's still in Sakexe_0

File: src/Network/Receive/ServerType0.pm [hunk 4]
```diff
@@ -2861,143 +2270,6 @@ sub character_equip {
...
-		}
-	} else {
-		#message "Storage password: unknown type $args->{type}\n";
-	}
-
-	# $args->{val}
-	# unknown, what is this for?
-}
-
-sub initialize_message_id_encryption {
```
Comment (lututui) [inline src/Network/Receive/ServerType0.pm]: initialize_message_id_encryption was removed from ServerType0 and added to Receive, but it's still in Sakexe_0

File: src/Network/Receive/ServerType0.pm [hunk 5]
```diff
@@ -3014,73 +2286,6 @@ sub top10 {
 	}
 }
 
-sub top10_alchemist_rank {
```
Comment (lututui) [inline src/Network/Receive/ServerType0.pm]: top10_alchemist_rank was removed from ServerType0 and added to Receive, but it's still in Sakexe_0

File: src/Network/Receive/ServerType0.pm [hunk 6]
```diff
@@ -3014,73 +2286,6 @@ sub top10 {
...
-	my ($self, $args) = @_;
-
-	my $textList = bytesToString(top10Listing($args));
-	message TF("============= ALCHEMIST RANK ================\n" .
-		"#    Name                             Points\n".
-		"%s" .
-		"=============================================\n", $textList), "list";
-}
-
-sub top10_blacksmith_rank {
```
Comment (lututui) [inline src/Network/Receive/ServerType0.pm]: top10_blacksmith_rank was removed from ServerType0 and added to Receive, but it's still in Sakexe_0

File: src/Network/Receive/ServerType0.pm [hunk 7]
```diff
@@ -3014,73 +2286,6 @@ sub top10 {
...
-	my ($self, $args) = @_;
-
-	my $textList = bytesToString(top10Listing($args));
-	message TF("============= BLACKSMITH RANK ===============\n" .
-		"#    Name                             Points\n".
-		"%s" .
-		"=============================================\n", $textList), "list";
-}
-
-sub top10_pk_rank {
```
Comment (lututui) [inline src/Network/Receive/ServerType0.pm]: top10_pk_rank was removed from ServerType0 and added to Receive, but it's still in Sakexe_0

File: src/Network/Receive/ServerType0.pm [hunk 8]
```diff
@@ -3014,73 +2286,6 @@ sub top10 {
...
-	my ($self, $args) = @_;
-
-	my $textList = bytesToString(top10Listing($args));
-	message TF("================ PVP RANK ===================\n" .
-		"#    Name                             Points\n".
-		"%s" .
-		"=============================================\n", $textList), "list";
-}
-
-sub top10_taekwon_rank {
```
Comment (lututui) [inline src/Network/Receive/ServerType0.pm]: top10_taekwon_rank was removed from ServerType0 and added to Receive, but it's still in Sakexe_0

File: src/Network/Receive/ServerType0.pm [hunk 9]
```diff
@@ -3014,73 +2286,6 @@ sub top10 {
...
-	my ($self, $args) = @_;
-
-	my $textList = bytesToString(top10Listing($args));
-	message TF("=============== TAEKWON RANK ================\n" .
-		"#    Name                             Points\n".
-		"%s" .
-		"=============================================\n", $textList), "list";
-}
-
-sub unequip_item {
```
Comment (lututui) [inline src/Network/Receive/ServerType0.pm]: unequip_item was removed from ServerType0 and added to Receive, but it's still in Sakexe_0

File: src/Network/Receive.pm [hunk 10]
```diff
@@ -7427,6 +7427,1260 @@ sub parse_merge_item_result {
...
+		}
+	}
+	if ($item) {
+		message TF("You unequip %s (%d) - %s\n",
+			$item->{name}, $item->{binID},
+			$equipTypes_lut{$item->{type_equip}}), 'inventory';
+	}
+}
+# TODO: only used to report failure? $args->{success}
+sub use_item {
```
Comment (lututui) [inline src/Network/Receive.pm]: use_item was added to Receive, but it's still in Sakexe_0 and ServerType0

Comment (lututui) [reply src/Network/Receive.pm]: Also, the packing in ServerType0 looks broken (`a2 x2 C`)

File: src/Network/Receive.pm [hunk 11]
```diff
@@ -7427,6 +7427,1260 @@ sub parse_merge_item_result {
...
+	my ($self, $args) = @_;
+	return unless changeToInGameState();
+	my $item = $char->inventory->getByID($args->{ID});
+	if ($item) {
+		message TF("You used Item: %s (%d) x %s\n", $item->{name}, $item->{binID}, $args->{amount}), "useItem";
+		inventoryItemRemoved($item->{binID}, $args->{amount});
+	}
+}
+
+sub users_online {
```
Comment (lututui) [inline src/Network/Receive.pm]: users_online was added to Receive, but it's still in Sakexe_0 and ServerType0

File: src/Network/Receive.pm [hunk 12]
```diff
@@ -7427,6 +7427,1260 @@ sub parse_merge_item_result {
...
+	my ($self, $args) = @_;
+	message TF("There are currently %s users online\n", $args->{users}), "info";
+}
+
+# You see a vender!  Add them to the visible venders list.
+sub vender_found {
```
Comment (lututui) [inline src/Network/Receive.pm]: vender_found was added to Receive, but it's still in Sakexe_0 and ServerType0

File: src/Network/Receive.pm [hunk 13]
```diff
@@ -7427,6 +7427,1260 @@ sub parse_merge_item_result {
...
+
+	if (!$venderLists{$ID} || !%{$venderLists{$ID}}) {
+		binAdd(\@venderListsID, $ID);
+		Plugins::callHook('packet_vender', {ID => $ID, title => bytesToString($args->{title})});
+	}
+	$venderLists{$ID}{title} = bytesToString($args->{title});
+	$venderLists{$ID}{id} = $ID;
+}
+
+sub vender_lost {
```
Comment (lututui) [inline src/Network/Receive.pm]: vender_lost was added to Receive, but it's still in Sakexe_0 and ServerType0

**Comment by ya4ept** [comment]
in such changes it is easy to implement backdoor 😈

**Comment by alisonrag** [comment]
this is why we review the code ^^

**Comment by Fadreus** [comment]
image

Hmmm... :thinking:

**Pull request merged**